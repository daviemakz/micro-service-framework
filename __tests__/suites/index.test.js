'use strict';

// Import NPM modules
import request from 'request';

// Import consts and functions
import {
  defaultServiceOptions,
  buildSecureOptions,
  buildHttpOptions,
  defaultInstanceOptions,
  CommandBodyObject,
  ResponseBodyObject,
  MicroServiceFramework
} from './../../dist';

// Hide console
// console.log = () => void 0;
// console.warn = () => void 0;

// Define variables
const sslOptions = {
  key: './__tests__/source/ssl/key.pem',
  cert: './__tests__/source/ssl/server.crt'
};

const httpOptions = {
  port: 12000,
  ssl: sslOptions,
  harden: true,
  middlewares: [],
  static: ['public'],
  routes: [
    {
      method: 'GET',
      uri: '/service',
      preMiddleware: [],
      postMiddleware: [],
      handler: (...args) => {
        return args;
      }
    },
    {
      method: 'GET',
      uri: '/'
    }
  ]
};

// Test suite
describe('src/index', () => {
  describe('const defaultInstanceOptions', () => {
    test('snapshot is expected', () => {
      expect(defaultInstanceOptions).toMatchSnapshot();
    });
  });
  describe('const defaultServiceOptions', () => {
    test('snapshot is expected', () => {
      expect(defaultServiceOptions).toMatchSnapshot();
    });
  });
  describe('buildSecureOptions()', () => {
    test('snapshot is expected', () => {
      expect(buildSecureOptions(sslOptions)).toMatchSnapshot();
    });
  });
  describe('buildHttpOptions()', () => {
    test('snapshot is expected', () => {
      expect(buildHttpOptions(httpOptions)).toMatchSnapshot();
    });
  });
  describe('ResponseBodyObject()', () => {
    test('snapshot is expected', () => {
      expect(ResponseBodyObject()).toMatchSnapshot();
    });
  });
  describe('CommandBodyObject()', () => {
    test('snapshot is expected', () => {
      expect(CommandBodyObject()).toMatchSnapshot();
    });
  });
  describe('class MicroServiceFramework()', () => {
    // Initialise instance
    const MSFrameworkInstance = new MicroServiceFramework({
      databaseNames: ['_defaultDatabase'],
      logPath: './server.log',
      mode: 'server',
      http: [
        {
          port: 12000,
          ssl: sslOptions,
          harden: true,
          beforeStart: express => express,
          middlewares: [],
          static: ['public'],
          routes: [
            {
              method: 'GET',
              uri: '/service',
              preMiddleware: [],
              postMiddleware: [],
              handler: (req, res, { sendRequest, CommandBodyObject }) => {
                // Define command body
                const testServiceCommandBody = new CommandBodyObject();
                // Assign command information
                testServiceCommandBody.destination = 'testService';
                testServiceCommandBody.funcName = 'testInfo';
                testServiceCommandBody.body = null;
                // Send initial request to micro service
                sendRequest(
                  testServiceCommandBody,
                  'testService',
                  false,
                  void 0,
                  void 0,
                  response => {
                    // Send back to response
                    const getPage = message => `
                    <html>
                       <head>
                          <title>Super Service Response</title>
                       </head>
                       <body>
                          <p>${message}</p>
                       </body>
                    </html>
                    `;
                    // Build HTML
                    const pageOutput = getPage(response);
                    // Return
                    res.send(pageOutput);
                  }
                );
              }
            }
          ]
        }
      ],
      runOnStart: [],
      verbose: true
    });

    // Define path to the operations file
    const testServerOperationsPath = './__tests__/source/services/testService';

    // Define a micro service
    MSFrameworkInstance.defineService('testService', testServerOperationsPath, {
      runOnStart: [],
      instances: 5
    });

    test('initialisation snapshot is expected', () => {
      expect(MSFrameworkInstance).toMatchSnapshot();
    });

    test('integration: micro service works', async () => {
      MSFrameworkInstance.startServer();
      let resultFromService = await new Promise(resolve => {
        setTimeout(
          () =>
            request('https://localhost:12000', {}, (err, res, body) =>
              resolve(body)
            ),
          5000
        );
      });
      expect(resultFromService).toMatchSnapshot();
    });
  });
});
