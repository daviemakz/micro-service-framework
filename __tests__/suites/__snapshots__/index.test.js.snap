// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/index CommandBodyObject() snapshot is expected 1`] = `
Object {
  "body": Object {},
  "destination": undefined,
  "funcName": "",
}
`;

exports[`src/index ResponseBodyObject() snapshot is expected 1`] = `
Object {
  "resultBody": Object {
    "errData": Object {},
    "resData": Object {},
  },
  "status": Object {
    "command": Object {
      "code": 100,
      "message": "Command completed successfully",
    },
    "transport": Object {
      "code": 1000,
      "message": "Transport completed successfully",
      "responseSource": "",
    },
  },
}
`;

exports[`src/index buildHttpOptions() snapshot is expected 1`] = `
Object {
  "beforeStart": [Function],
  "harden": true,
  "middlewares": Array [],
  "port": 12000,
  "routes": Array [
    Object {
      "handler": [Function],
      "method": "GET",
      "postMiddleware": Array [],
      "preMiddleware": Array [],
      "uri": "/service",
    },
    Object {
      "method": "GET",
      "uri": "/",
    },
  ],
  "ssl": Object {
    "cert": "-----BEGIN CERTIFICATE-----
MIIDpDCCAowCCQDZhz2bUBip1jANBgkqhkiG9w0BAQsFADCBkzELMAkGA1UEBhMC
R0IxDzANBgNVBAgMBkxvbmRvbjEPMA0GA1UEBwwGTG9uZG9uMRMwEQYDVQQKDApN
YWt1bmkgTFRkMRUwEwYDVQQLDAxXZWIgU2VydmljZXMxEjAQBgNVBAMMCWxvY2Fs
aG9zdDEiMCAGCSqGSIb3DQEJARYTZC5tYWt1bmlAbGl2ZS5jby51azAeFw0xOTAy
MTEyMjE5MDFaFw0yMDAyMTEyMjE5MDFaMIGTMQswCQYDVQQGEwJHQjEPMA0GA1UE
CAwGTG9uZG9uMQ8wDQYDVQQHDAZMb25kb24xEzARBgNVBAoMCk1ha3VuaSBMVGQx
FTATBgNVBAsMDFdlYiBTZXJ2aWNlczESMBAGA1UEAwwJbG9jYWxob3N0MSIwIAYJ
KoZIhvcNAQkBFhNkLm1ha3VuaUBsaXZlLmNvLnVrMIIBIjANBgkqhkiG9w0BAQEF
AAOCAQ8AMIIBCgKCAQEAx7LxiJOxAtDI9F/bEkcBXfDv7ts5cKOd8s+/LwQB/BnQ
Kc6vnaqQAHbwrPPo4QF4fsaT7hjOMFx2B3dAHLAhrQ2iy+jHaUdvTtcxthOGK6ag
HnlEPfsLCwIBGHKR7Eb/niuMNabQS27mPEDOrt1orBBaTxLWWeC5pXvz7ezp2ZVF
YeqBKkUP00QBYH1MoB3YuoU5swtx+AZiiFkEKMUa+JoKbN1D3wFWbo1GesMfSf/x
Lrvx6qqed11JOCa2YNpkxaKD+SwCzhAkh9xqdoOUfFOVlQNlMX7wwGR5fqDQypw7
2foqp29nqKgJR1Hgteb627N2EQT+ZnhhQK2GCSqVOQIDAQABMA0GCSqGSIb3DQEB
CwUAA4IBAQCvwJvIlq4XeadDR82a/LbGH2rBbHfmdQtOC4dhbJHbmvwqp/YI7DQ0
L+69zeZPQNwUu0t5BBtR75II7MvZclga0w2BYE+sdhHDcoB03teuBM6IgfaGt0tA
BcaWxZ7WkN0Py5n+59ASQtymd+BPghH6G+toDjrBvF1Yhg1OLZlA0M9w/Ekcph8h
TP5aoCN4TipTOUZ+P/ROcStyReC3enysRMBoN1GrBATxU0VxSaLd5AT5Ztq44q5v
upb39HZvD9aPaWGxG8ODtF5BHH/MJW5PAtuwz5IbxlBg7qTMN1BYf/W8BCMnrhS7
8qu1PzL1L1T9xPeeycPhBT1Z1ewtfLw2
-----END CERTIFICATE-----
",
    "key": "-----BEGIN PRIVATE KEY-----
MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDHsvGIk7EC0Mj0
X9sSRwFd8O/u2zlwo53yz78vBAH8GdApzq+dqpAAdvCs8+jhAXh+xpPuGM4wXHYH
d0AcsCGtDaLL6MdpR29O1zG2E4YrpqAeeUQ9+wsLAgEYcpHsRv+eK4w1ptBLbuY8
QM6u3WisEFpPEtZZ4Lmle/Pt7OnZlUVh6oEqRQ/TRAFgfUygHdi6hTmzC3H4BmKI
WQQoxRr4mgps3UPfAVZujUZ6wx9J//Euu/Hqqp53XUk4JrZg2mTFooP5LALOECSH
3Gp2g5R8U5WVA2UxfvDAZHl+oNDKnDvZ+iqnb2eoqAlHUeC15vrbs3YRBP5meGFA
rYYJKpU5AgMBAAECggEBAJqJo9hpUqY3JAvY0hNG14v3E/XnuH+uXmC02+JqW47X
+vsfc0+z2lRBtuZgqmrVVIXm7jtm/sj2BHCqm/aXZrggwUbDEO4r1btRVKruWPoW
q/ppJaJBT75pAqgEoWATYWTplks+mBqH66vdCxTPSl09GmXu0eojugaR8abBnAY/
2n7CbUvwzhoT2h/sJx+TJji4b42yPVuyJlbBmegSjDADCWEzXVIuiwOfPJLwLzE3
l9YszK7pYo7FZMk0Jb3r1CnAIp3p3YKILfSqpFeKVw/wnlB1kTeskkdd+HWVZo4A
ISHXfAdcCxmevxmvsDecCE9AGYsa7BNa/UfBr2nVo8ECgYEA9m38XtOeFzb6phlz
TQS0iPKbdW2S7hNwl6uRAaCxiqYO7aLA/1xmmvHPmtLoktS5Ypdk9gycCu9+axlw
MDKSOrN+dAd5zxRgtLfuzeR9YdXpEhFcYq/UruRJfMuAB3bFi5kmdm0tKKa3ptBr
zCDlz5L41qoqpQgDKmSICCdOb7UCgYEAz3RcErgQA49A6m6Pr9/tff+bOQtpuFbJ
gtty8UKJaqIjxMGQ4JkT2fHBfFE7JrWRyGis7ATfoAcxxgoSvbxRFAOz8Md+qF2v
6zwWbrS4AXQfSpc89XolIQedj/Q2yzVREdiO539+Li4gox0JCpdTQ/ssfsHUo6PB
9OHtH34NGfUCgYBZ+R5DoQ//edtZ+cDXRfiEZ8/KuA8FTaO2PLXc2kb4UFjy+pFs
qtfrio46APdZZMGq9YuvKjtu4rJNeCQ19MDfHdn1kyUuLkqAsQdHtTzyE5/dMOmC
PqBwmnkaT1KXZw4JQgmBA6uRRA2Zok85YGmHuCmSdt7LEnnVaI8iKvfAYQKBgGHB
cQmZFmK3BF3zz12O5Xh0FA7jY+PrcXVqN1WchuhqkvRKShyrxYDzvLr6BCePc6y+
No3ZFg9wlHG/mWky7fnCbO5O804fJ6BvZhqwqex1bbFWZvE9p3VTA1osOA21iAgi
xeQsXachMVvreLE28ZOdLjBk4fBiYCkmtPgcC8xpAoGAdzJhJ+42tKDkqNuDEMYg
3DUff+GMfFpXOv/kB83Ljp6G93Y3filjxIBGgU9RuOOSs63ajW5T8m1sSZPtJZWS
5u85aLAeFpr1N5fRsNgL9tD7S17dyAYm190RWRWWr8XZ4FEi8YvABf4Sj/2YXCAN
mO7W51tU6ExL/VQfqjpSRPE=
-----END PRIVATE KEY-----
",
  },
  "static": Array [
    "public",
  ],
}
`;

exports[`src/index buildSecureOptions() snapshot is expected 1`] = `
Object {
  "cert": "-----BEGIN CERTIFICATE-----
MIIDpDCCAowCCQDZhz2bUBip1jANBgkqhkiG9w0BAQsFADCBkzELMAkGA1UEBhMC
R0IxDzANBgNVBAgMBkxvbmRvbjEPMA0GA1UEBwwGTG9uZG9uMRMwEQYDVQQKDApN
YWt1bmkgTFRkMRUwEwYDVQQLDAxXZWIgU2VydmljZXMxEjAQBgNVBAMMCWxvY2Fs
aG9zdDEiMCAGCSqGSIb3DQEJARYTZC5tYWt1bmlAbGl2ZS5jby51azAeFw0xOTAy
MTEyMjE5MDFaFw0yMDAyMTEyMjE5MDFaMIGTMQswCQYDVQQGEwJHQjEPMA0GA1UE
CAwGTG9uZG9uMQ8wDQYDVQQHDAZMb25kb24xEzARBgNVBAoMCk1ha3VuaSBMVGQx
FTATBgNVBAsMDFdlYiBTZXJ2aWNlczESMBAGA1UEAwwJbG9jYWxob3N0MSIwIAYJ
KoZIhvcNAQkBFhNkLm1ha3VuaUBsaXZlLmNvLnVrMIIBIjANBgkqhkiG9w0BAQEF
AAOCAQ8AMIIBCgKCAQEAx7LxiJOxAtDI9F/bEkcBXfDv7ts5cKOd8s+/LwQB/BnQ
Kc6vnaqQAHbwrPPo4QF4fsaT7hjOMFx2B3dAHLAhrQ2iy+jHaUdvTtcxthOGK6ag
HnlEPfsLCwIBGHKR7Eb/niuMNabQS27mPEDOrt1orBBaTxLWWeC5pXvz7ezp2ZVF
YeqBKkUP00QBYH1MoB3YuoU5swtx+AZiiFkEKMUa+JoKbN1D3wFWbo1GesMfSf/x
Lrvx6qqed11JOCa2YNpkxaKD+SwCzhAkh9xqdoOUfFOVlQNlMX7wwGR5fqDQypw7
2foqp29nqKgJR1Hgteb627N2EQT+ZnhhQK2GCSqVOQIDAQABMA0GCSqGSIb3DQEB
CwUAA4IBAQCvwJvIlq4XeadDR82a/LbGH2rBbHfmdQtOC4dhbJHbmvwqp/YI7DQ0
L+69zeZPQNwUu0t5BBtR75II7MvZclga0w2BYE+sdhHDcoB03teuBM6IgfaGt0tA
BcaWxZ7WkN0Py5n+59ASQtymd+BPghH6G+toDjrBvF1Yhg1OLZlA0M9w/Ekcph8h
TP5aoCN4TipTOUZ+P/ROcStyReC3enysRMBoN1GrBATxU0VxSaLd5AT5Ztq44q5v
upb39HZvD9aPaWGxG8ODtF5BHH/MJW5PAtuwz5IbxlBg7qTMN1BYf/W8BCMnrhS7
8qu1PzL1L1T9xPeeycPhBT1Z1ewtfLw2
-----END CERTIFICATE-----
",
  "key": "-----BEGIN PRIVATE KEY-----
MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDHsvGIk7EC0Mj0
X9sSRwFd8O/u2zlwo53yz78vBAH8GdApzq+dqpAAdvCs8+jhAXh+xpPuGM4wXHYH
d0AcsCGtDaLL6MdpR29O1zG2E4YrpqAeeUQ9+wsLAgEYcpHsRv+eK4w1ptBLbuY8
QM6u3WisEFpPEtZZ4Lmle/Pt7OnZlUVh6oEqRQ/TRAFgfUygHdi6hTmzC3H4BmKI
WQQoxRr4mgps3UPfAVZujUZ6wx9J//Euu/Hqqp53XUk4JrZg2mTFooP5LALOECSH
3Gp2g5R8U5WVA2UxfvDAZHl+oNDKnDvZ+iqnb2eoqAlHUeC15vrbs3YRBP5meGFA
rYYJKpU5AgMBAAECggEBAJqJo9hpUqY3JAvY0hNG14v3E/XnuH+uXmC02+JqW47X
+vsfc0+z2lRBtuZgqmrVVIXm7jtm/sj2BHCqm/aXZrggwUbDEO4r1btRVKruWPoW
q/ppJaJBT75pAqgEoWATYWTplks+mBqH66vdCxTPSl09GmXu0eojugaR8abBnAY/
2n7CbUvwzhoT2h/sJx+TJji4b42yPVuyJlbBmegSjDADCWEzXVIuiwOfPJLwLzE3
l9YszK7pYo7FZMk0Jb3r1CnAIp3p3YKILfSqpFeKVw/wnlB1kTeskkdd+HWVZo4A
ISHXfAdcCxmevxmvsDecCE9AGYsa7BNa/UfBr2nVo8ECgYEA9m38XtOeFzb6phlz
TQS0iPKbdW2S7hNwl6uRAaCxiqYO7aLA/1xmmvHPmtLoktS5Ypdk9gycCu9+axlw
MDKSOrN+dAd5zxRgtLfuzeR9YdXpEhFcYq/UruRJfMuAB3bFi5kmdm0tKKa3ptBr
zCDlz5L41qoqpQgDKmSICCdOb7UCgYEAz3RcErgQA49A6m6Pr9/tff+bOQtpuFbJ
gtty8UKJaqIjxMGQ4JkT2fHBfFE7JrWRyGis7ATfoAcxxgoSvbxRFAOz8Md+qF2v
6zwWbrS4AXQfSpc89XolIQedj/Q2yzVREdiO539+Li4gox0JCpdTQ/ssfsHUo6PB
9OHtH34NGfUCgYBZ+R5DoQ//edtZ+cDXRfiEZ8/KuA8FTaO2PLXc2kb4UFjy+pFs
qtfrio46APdZZMGq9YuvKjtu4rJNeCQ19MDfHdn1kyUuLkqAsQdHtTzyE5/dMOmC
PqBwmnkaT1KXZw4JQgmBA6uRRA2Zok85YGmHuCmSdt7LEnnVaI8iKvfAYQKBgGHB
cQmZFmK3BF3zz12O5Xh0FA7jY+PrcXVqN1WchuhqkvRKShyrxYDzvLr6BCePc6y+
No3ZFg9wlHG/mWky7fnCbO5O804fJ6BvZhqwqex1bbFWZvE9p3VTA1osOA21iAgi
xeQsXachMVvreLE28ZOdLjBk4fBiYCkmtPgcC8xpAoGAdzJhJ+42tKDkqNuDEMYg
3DUff+GMfFpXOv/kB83Ljp6G93Y3filjxIBGgU9RuOOSs63ajW5T8m1sSZPtJZWS
5u85aLAeFpr1N5fRsNgL9tD7S17dyAYm190RWRWWr8XZ4FEi8YvABf4Sj/2YXCAN
mO7W51tU6ExL/VQfqjpSRPE=
-----END PRIVATE KEY-----
",
}
`;

exports[`src/index class MicroServiceFramework() initialisation snapshot is expected 1`] = `
MicroServiceFramework {
  "addServerToTracking": [Function],
  "assignCoreFunctions": [Function],
  "bindGateway": [Function],
  "checkConnection": [Function],
  "conId": 0,
  "coreOperations": Object {},
  "databaseOperation": [Function],
  "db": Object {
    "_defaultDatabase": table {
      "add": [Function],
      "all": [Function],
      "delete": [Function],
      "fetch": [Function],
      "fetchAll": [Function],
      "get": [Function],
      "has": [Function],
      "includes": [Function],
      "push": [Function],
      "set": [Function],
      "startsWith": [Function],
      "subtract": [Function],
      "tableName": "_defaultDatabase",
    },
  },
  "destinationUnknown": [Function],
  "destroyConnection": [Function],
  "eventHandlers": Object {
    "onConClose": [Function],
    "onConRequest": [Function],
  },
  "executeInitialFunctions": [Function],
  "externalInterfaces": Object {},
  "functionUnknown": [Function],
  "hardenServer": [Function],
  "httpServer": Array [],
  "httpsServer": Array [],
  "inUsePorts": Array [],
  "initConnectionToService": [Function],
  "initGateway": [Function],
  "initService": [Function],
  "initiateMicroServerConnection": [Function],
  "invokeListener": [Function],
  "invokeSpeaker": [Function],
  "log": [Function],
  "microServerCommunication": [Function],
  "microServiceStarted": false,
  "processComError": [Function],
  "processComRequest": [Function],
  "removeServerFromTracking": [Function],
  "sendRequest": [Function],
  "sentReplyToSocket": [Function],
  "serviceData": Object {},
  "serviceInfo": Object {},
  "serviceOptions": Object {},
  "settings": Object {
    "apiGatewayPort": 8080,
    "connectionTimeout": 1000,
    "coreOperations": Object {
      "runScript": [Function],
    },
    "databaseNames": Array [
      "_defaultDatabase",
    ],
    "http": Array [
      Object {
        "beforeStart": [Function],
        "harden": true,
        "middlewares": Array [],
        "port": 12000,
        "routes": Array [
          Object {
            "handler": [Function],
            "method": "GET",
            "postMiddleware": Array [],
            "preMiddleware": Array [],
            "uri": "/service",
          },
        ],
        "ssl": Object {
          "cert": "-----BEGIN CERTIFICATE-----
MIIDpDCCAowCCQDZhz2bUBip1jANBgkqhkiG9w0BAQsFADCBkzELMAkGA1UEBhMC
R0IxDzANBgNVBAgMBkxvbmRvbjEPMA0GA1UEBwwGTG9uZG9uMRMwEQYDVQQKDApN
YWt1bmkgTFRkMRUwEwYDVQQLDAxXZWIgU2VydmljZXMxEjAQBgNVBAMMCWxvY2Fs
aG9zdDEiMCAGCSqGSIb3DQEJARYTZC5tYWt1bmlAbGl2ZS5jby51azAeFw0xOTAy
MTEyMjE5MDFaFw0yMDAyMTEyMjE5MDFaMIGTMQswCQYDVQQGEwJHQjEPMA0GA1UE
CAwGTG9uZG9uMQ8wDQYDVQQHDAZMb25kb24xEzARBgNVBAoMCk1ha3VuaSBMVGQx
FTATBgNVBAsMDFdlYiBTZXJ2aWNlczESMBAGA1UEAwwJbG9jYWxob3N0MSIwIAYJ
KoZIhvcNAQkBFhNkLm1ha3VuaUBsaXZlLmNvLnVrMIIBIjANBgkqhkiG9w0BAQEF
AAOCAQ8AMIIBCgKCAQEAx7LxiJOxAtDI9F/bEkcBXfDv7ts5cKOd8s+/LwQB/BnQ
Kc6vnaqQAHbwrPPo4QF4fsaT7hjOMFx2B3dAHLAhrQ2iy+jHaUdvTtcxthOGK6ag
HnlEPfsLCwIBGHKR7Eb/niuMNabQS27mPEDOrt1orBBaTxLWWeC5pXvz7ezp2ZVF
YeqBKkUP00QBYH1MoB3YuoU5swtx+AZiiFkEKMUa+JoKbN1D3wFWbo1GesMfSf/x
Lrvx6qqed11JOCa2YNpkxaKD+SwCzhAkh9xqdoOUfFOVlQNlMX7wwGR5fqDQypw7
2foqp29nqKgJR1Hgteb627N2EQT+ZnhhQK2GCSqVOQIDAQABMA0GCSqGSIb3DQEB
CwUAA4IBAQCvwJvIlq4XeadDR82a/LbGH2rBbHfmdQtOC4dhbJHbmvwqp/YI7DQ0
L+69zeZPQNwUu0t5BBtR75II7MvZclga0w2BYE+sdhHDcoB03teuBM6IgfaGt0tA
BcaWxZ7WkN0Py5n+59ASQtymd+BPghH6G+toDjrBvF1Yhg1OLZlA0M9w/Ekcph8h
TP5aoCN4TipTOUZ+P/ROcStyReC3enysRMBoN1GrBATxU0VxSaLd5AT5Ztq44q5v
upb39HZvD9aPaWGxG8ODtF5BHH/MJW5PAtuwz5IbxlBg7qTMN1BYf/W8BCMnrhS7
8qu1PzL1L1T9xPeeycPhBT1Z1ewtfLw2
-----END CERTIFICATE-----
",
          "key": "-----BEGIN PRIVATE KEY-----
MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDHsvGIk7EC0Mj0
X9sSRwFd8O/u2zlwo53yz78vBAH8GdApzq+dqpAAdvCs8+jhAXh+xpPuGM4wXHYH
d0AcsCGtDaLL6MdpR29O1zG2E4YrpqAeeUQ9+wsLAgEYcpHsRv+eK4w1ptBLbuY8
QM6u3WisEFpPEtZZ4Lmle/Pt7OnZlUVh6oEqRQ/TRAFgfUygHdi6hTmzC3H4BmKI
WQQoxRr4mgps3UPfAVZujUZ6wx9J//Euu/Hqqp53XUk4JrZg2mTFooP5LALOECSH
3Gp2g5R8U5WVA2UxfvDAZHl+oNDKnDvZ+iqnb2eoqAlHUeC15vrbs3YRBP5meGFA
rYYJKpU5AgMBAAECggEBAJqJo9hpUqY3JAvY0hNG14v3E/XnuH+uXmC02+JqW47X
+vsfc0+z2lRBtuZgqmrVVIXm7jtm/sj2BHCqm/aXZrggwUbDEO4r1btRVKruWPoW
q/ppJaJBT75pAqgEoWATYWTplks+mBqH66vdCxTPSl09GmXu0eojugaR8abBnAY/
2n7CbUvwzhoT2h/sJx+TJji4b42yPVuyJlbBmegSjDADCWEzXVIuiwOfPJLwLzE3
l9YszK7pYo7FZMk0Jb3r1CnAIp3p3YKILfSqpFeKVw/wnlB1kTeskkdd+HWVZo4A
ISHXfAdcCxmevxmvsDecCE9AGYsa7BNa/UfBr2nVo8ECgYEA9m38XtOeFzb6phlz
TQS0iPKbdW2S7hNwl6uRAaCxiqYO7aLA/1xmmvHPmtLoktS5Ypdk9gycCu9+axlw
MDKSOrN+dAd5zxRgtLfuzeR9YdXpEhFcYq/UruRJfMuAB3bFi5kmdm0tKKa3ptBr
zCDlz5L41qoqpQgDKmSICCdOb7UCgYEAz3RcErgQA49A6m6Pr9/tff+bOQtpuFbJ
gtty8UKJaqIjxMGQ4JkT2fHBfFE7JrWRyGis7ATfoAcxxgoSvbxRFAOz8Md+qF2v
6zwWbrS4AXQfSpc89XolIQedj/Q2yzVREdiO539+Li4gox0JCpdTQ/ssfsHUo6PB
9OHtH34NGfUCgYBZ+R5DoQ//edtZ+cDXRfiEZ8/KuA8FTaO2PLXc2kb4UFjy+pFs
qtfrio46APdZZMGq9YuvKjtu4rJNeCQ19MDfHdn1kyUuLkqAsQdHtTzyE5/dMOmC
PqBwmnkaT1KXZw4JQgmBA6uRRA2Zok85YGmHuCmSdt7LEnnVaI8iKvfAYQKBgGHB
cQmZFmK3BF3zz12O5Xh0FA7jY+PrcXVqN1WchuhqkvRKShyrxYDzvLr6BCePc6y+
No3ZFg9wlHG/mWky7fnCbO5O804fJ6BvZhqwqex1bbFWZvE9p3VTA1osOA21iAgi
xeQsXachMVvreLE28ZOdLjBk4fBiYCkmtPgcC8xpAoGAdzJhJ+42tKDkqNuDEMYg
3DUff+GMfFpXOv/kB83Ljp6G93Y3filjxIBGgU9RuOOSs63ajW5T8m1sSZPtJZWS
5u85aLAeFpr1N5fRsNgL9tD7S17dyAYm190RWRWWr8XZ4FEi8YvABf4Sj/2YXCAN
mO7W51tU6ExL/VQfqjpSRPE=
-----END PRIVATE KEY-----
",
        },
        "static": Array [
          "public",
        ],
      },
    ],
    "logPath": "./server.log",
    "maxBuffer": 50,
    "microServiceConnectionAttempts": 1000,
    "microServiceConnectionTimeout": 10000,
    "mode": "server",
    "onConClose": [Function],
    "onConRequest": [Function],
    "portRangeFinish": 65535,
    "portRangeStart": 1024,
    "restartTimeout": 50,
    "runOnStart": Array [],
    "verbose": true,
  },
  "startHttpServer": [Function],
  "startServer": [Function],
  "startServerFailed": [Function],
}
`;

exports[`src/index const defaultInstanceOptions snapshot is expected 1`] = `
Object {
  "apiGatewayPort": 8080,
  "connectionTimeout": 1000,
  "coreOperations": Object {},
  "databaseNames": Array [
    "_defaultTable",
  ],
  "http": false,
  "logPath": undefined,
  "maxBuffer": 50,
  "microServiceConnectionAttempts": 1000,
  "microServiceConnectionTimeout": 10000,
  "mode": "server",
  "portRangeFinish": 65535,
  "portRangeStart": 1024,
  "restartTimeout": 50,
  "runOnStart": Array [],
  "verbose": true,
}
`;

exports[`src/index const defaultServiceOptions snapshot is expected 1`] = `
Object {
  "instances": 1,
  "loadBalancing": "roundRobin",
  "runOnStart": Array [],
}
`;
